### PR TITLE
INS-16245 redis cdc resize via terraform

### PR DESCRIPTION
**Pull request by weixiong-ins**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ test/mock_api_server/__pycache__
 
 vendor
 .idea
+.vscode
```
Comment (charlese-instaclustr) [inline .gitignore:6]: Intentional change?

File: Makefile [hunk 2]
```diff
@@ -3,7 +3,7 @@ BIN_NAME=terraform-provider-instaclustr
 
 
 # for VERSION, don't add prefix "v", e.g., use "1.9.8" instead of "v1.9.8" as it could break circleCI stuff
-VERSION=1.14.4
+VERSION=1.14.5
```
Comment (charlese-instaclustr) [inline Makefile:6]: Should this be a minor version bump since we are adding functionality rather than patching? So `1.15.0`

File: acc_test/data/valid_with_resizable_redis_cluster.tf [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  cluster_network         = "192.168.0.0/18"
+  private_network_cluster = false
+  pci_compliant_cluster   = false
+  wait_for_state          = "RUNNING"
+  cluster_provider = {
+    name = "AWS_VPC"
+  }
+  bundle {
+    bundle  = "REDIS"
+    version = "redis:6.0.9"
```
Comment (charlese-instaclustr) [inline acc_test/data/valid_with_resizable_redis_cluster.tf:21]: Might as well use our new `6.2.6` release here, I suppose?

File: instaclustr/resource_cluster_unit_test.go [hunk 4]
```diff
@@ -499,11 +499,11 @@ func TestHasCassandraSizeChanges(t *testing.T) {
 func TestDoClusterResizeDefault(t *testing.T) {
 	err := doClusterResize(MockApiClient{
 		cluster: Cluster{
-			ID:         "REDIS",
-			BundleType: "REDIS",
+			ID:         "CADENCE",
```
Comment (michael-carter-instaclustr) [inline instaclustr/resource_cluster_unit_test.go:502]: Are these changes intentional? What does CADENCE have to do with resizing REDIS?

File: instaclustr/resource_cluster.go [hunk 5]
```diff
@@ -1181,6 +1193,40 @@ func doLegacyCassandraClusterResize(client APIClientInterface, cluster *Cluster,
 	return nil
 }
 
+func isRedisSizeAllChange(nodeSize string) (string, bool) {
```
Comment (michael-carter-instaclustr) [inline instaclustr/resource_cluster.go:1196]: Is this function used anywhere?

File: acc_test/resource_cluster_test.go [hunk 6]
```diff
@@ -363,6 +363,51 @@ func TestAccOpenSearchClusterResize(t *testing.T) {
 	})
 }
 
+func TestAccRedisClusterResize(t *testing.T) {
```
Comment (michael-carter-instaclustr) [inline acc_test/resource_cluster_test.go:366]: The acc tests are slow and flakey at the moment, largely because there are a bunch of tests that wait until a cluster reaches running, which takes a long time.
Could you please make sure that you aren't creating more REDIS clusters than is absolutely necessary. i.e. If there are more than one test cases that make REDIS clusters and wait for them to reach running, consider consolidating them (Yu has done this for Kafka in the other open PR at the moment).

File: Makefile (hunk 2 above)
Comment (michael-carter-instaclustr) [inline Makefile:6]: You'll be competing with Yu for this version number. It may need to be changed depending on who merges first

File: instaclustr/resource_cluster_unit_test.go [hunk 7]
```diff
@@ -499,11 +499,11 @@ func TestHasCassandraSizeChanges(t *testing.T) {
...
+			BundleType: "CADENCE",
```
Comment (charlese-instaclustr) [inline instaclustr/resource_cluster_unit_test.go:503]: Why are we changing these values to CADENCE from REDIS?

**Review by charlese-instaclustr — CHANGES_REQUESTED** [review]

File: instaclustr/resource_cluster_unit_test.go (hunk 4 above)
Comment (weixiong-ins) [reply instaclustr/resource_cluster_unit_test.go:502]: Yes, these changes are intentional. It will test the default behaviour when users want to perform resize of the bundle we haven't supported yet. Previously its Redis, now hand over to CADENCE.

File: instaclustr/resource_cluster.go (hunk 5 above)
Comment (weixiong-ins) [reply instaclustr/resource_cluster.go]: I have removed it in new commit.

File: acc_test/resource_cluster_test.go (hunk 6 above)
Comment (weixiong-ins) [reply acc_test/resource_cluster_test.go]: I have combined Redis tests together in "resource_redis_test.go"(in new commit).
Both "cluster create" and "resize" tests requires running status, they are merged into one case to save time.

File: .gitignore (hunk 1 above)
Comment (weixiong-ins) [reply .gitignore:6]: yes, I use vscode to develop. This may help others using vscode.

File: acc_test/data/valid_with_resizable_redis_cluster.tf (hunk 3 above)
Comment (weixiong-ins) [reply acc_test/data/valid_with_resizable_redis_cluster.tf]: currently maybe 6.0.9 is enough.

File: acc_test/resource_redis_test.go [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+					resource.TestCheckResourceAttr("instaclustr_cluster."+resourceName, "node_size", "t3.medium-80-r"),
+					testCheckClusterResize(resourceName, hostname, username, apiKey, "t3.medium-80-r"),
+				),
+			},
+		},
+	})
+}
+
+// Test that the options does re-create the Redis cluster
+// Disabling for now as it's failing for an unknown reason, blocking acc tests passing and isn't seen as terribly important to REDIS
```
Comment (charlese-instaclustr) [inline acc_test/resource_redis_test.go:71]: Do we need to hang onto this method since it `isn't seen as terribly important to REDIS` ? How long ago was this disabled?

Comment (weixiong-ins) [reply acc_test/resource_redis_test.go:71]: I didn't check this method in details actually. just move it from "resource_cluster_test.go" to "resource_redis_test.go", and keep the same.

Comment (weixiong-ins) [reply acc_test/resource_redis_test.go:71]: @charlese-instaclustr : discussed with @WeiXinFam-Insta, i'll remove it in my new commit.

**Comment by charlese-instaclustr** [comment]
LGTM

**Review by charlese-instaclustr — APPROVED** [review]

**Review by michael-carter-instaclustr — APPROVED** [review]
